gems/quilt_rails/README.md

```diff
@@ -215,18 +217,205 @@ class ReactController < ApplicationController
+
+```ruby
+client.on_report do |report, tags|
+  client.distribution('some_metric', some_computation(report), tags)
+end
+```
+
+##### Client#on_navigation
+
+The `on_report` method takes a block which is run once per navigation in a given `Report`. It can be used to perform side-effects based on a given `Navigation`, including sending custom `distribution`s.
+
+```ruby
+client.on_navigation do |navigation, tags|
+  weight = navigation.events_with_size.reduce(0) do |total, event|
+    total + event.size
+  end
+
+  client.distribution('navigation_total_resource_weight', weight, tags)
+end
+```
+
+##### Client#customize_tags
+
+The `customize_tags` method allows custom tags to be computed to include in every distribution for a given POST to the controller.
+
+```ruby
+client.customize_tags do |metrics_body, tags|
+  # pathname will be inluded on *every* distribution for this report
+  tags[:pathname] = metrics_body.pathname
+end
+```
+
+##### Client#customize_navigation_tags
+
+The `customize_navigation_tags` method allows custom tags to be computed for each individual navigation described by the POST body, and included on the `distribution` calls for that navigation.
+
+```ruby
+client.customize_navigation_tags do |navigation, tags|
+  # target will be included only for the distributions about this navigation
+  tags[:navigation_target] = navigation.target
+end
+```
+
+#### Report
+
+[`Quilt::Performance::Report`](/lib/quilt_rails/performance/report.rb#L3) describes timing, size information, and metadata about a series of events and navigations on the front-end of a web application, as recorded by `@shopify/performance` and parsed from an `ActionController:Parameters` instance.
+
+##### Report#pathname
+
+The initial path (eg. `/foo`) of the request which generated the report.
+
+##### Report#connection
+
+Information parsed from [`navigator.connection`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/connection) detailing the connectivity of the browser used to initiate the request.
+
+##### Report#events
+
+[`Quilt::Performance::Event`](/lib/quilt_rails/performance/event.rb#L3) instances containing details about milestones in the course of the request.
+
+##### Report#navigations
+
+[`Quilt::Performance::Navigation`](/lib/quilt_rails/performance/navigation.rb#L3) instances containing details about individual front-end navigations which took place as part of the request the report was generated for.
+
 ### Engine
 
-`Quilt::Engine` provides a preconfigured controller which consumes `ReactRenderable` and provides an index route which uses it.
+`Quilt::Engine` provides:
+
+- a preconfigured `UiController` which consumes `ReactRenderable`
+- a preconfigured `PerformanceController` which consumes `Performance::Reportable`
+- a `/performance` route mapped to `PerformanceController#index`
+- a catch-all index route mapped to the `UiController#index`
```

Comment:
Is there any way we can give them only the performance bit, or would that require them to manually use the mixin?

Reply:
manually use the mixin